### PR TITLE
refactor: color logic performance

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

[Description of the change]

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by JanDeDobbeleer** [comment]
@lnu I always disliked how the colorer worked (mainly trial and error development) so I took the time to make it cleaner and more performant by not repeating logic. Looks a lot more understandable now.

File: themes/bubbles.omp.json [hunk 1]
```diff
@@ -176,16 +177,6 @@
               "text": "\u276F",
               "postfix": ""
           }
-        },
-        {
```
Comment (JanDeDobbeleer) [inline themes/bubbles.omp.json:180]: This was a weird, a coincidentally supported transparent text segment. Not intended to work and should break.

**Comment by lnu** [comment]
> @lnu I always disliked how the colorer worked (mainly trial and error development) so I took the time to make it cleaner and more performant by not repeating logic. Looks a lot more understandable now.

The kind of stuff you’re glad to have unit tests for

**Comment by JanDeDobbeleer** [comment]
@lnu yes, although these could be better tbh (my own fault)

**Pull request merged**